pkg/issuer/venafi/client/request.go

```diff
@@ -66,9 +66,12 @@ func (v *Venafi) RequestCertificate(csrPEM []byte, duration time.Duration, custo
 	//
 	// Note that resetting won't affect the existing certificate if one was
 	// already issued.
-	tppConnector, isTPP := v.vcertClient.(*tpp.Connector)
-	if isTPP {
-		err := tppConnector.ResetCertificate(vreq, false)
+	if v.tppClient != nil {
+		// We can't use the instrumented v.vcertClient because its concrete
+		// value is `instrumentedConnector`, which doesn't give access to the
+		// *tpp.Connector it wraps. Also, `instrumentedConnector` doesn't
+		// support `ResetCertificate`.
+		err := v.tppClient.ResetCertificate(vreq, false)
```

Comment:
@maelvls it is still important to differentiate between VaaS and TPP, if you are connected to VaaS, you will get an error when you call `ResetCertificate`.

Reply:
Isn't it sufficient to gate this with `if v.tppClient != nil`?

Reply:
Yes, sorry, I misread.